dynamiq/nodes/llms/base.py

```diff
@@ -98,13 +95,42 @@ class BaseLLM(ConnectionNode):
     tool_choice: str | None = None
     thinking_enabled: bool | None = None
     budget_tokens: int = 1024
+    response_format: dict[str, Any] | None = None
+    tools: list[Tool] | None = None
+    _completion: Callable = PrivateAttr()
+    _stream_chunk_builder: Callable = PrivateAttr()
+    input_schema: ClassVar[type[BaseLLMInputSchema]] = BaseLLMInputSchema
     inference_mode: InferenceMode = InferenceMode.DEFAULT
```

Comment:
For the params that we plan to deprecate, let's add deprecated helper like:
`inference_mode: InferenceMode = Field(default=InferenceMode.DEFAULT, deprecated="Please use <new params> param/s instead")`